go/vrf/ecvrf.go

```diff
@@ -49,14 +60,23 @@ func NewKey(curve elliptic.Curve, sk []byte) *PrivateKey {
 	}
 }
 
+type ECVRF interface {
+	Params() *ECVRFParams
+
+	// Prove returns proof pi that beta is the correct hash output.
+	// beta is deterministic in the sense that it always
+	// produces the same output beta given a pair of inputs (sk, alpha).
+	Prove(sk *PrivateKey, alpha []byte) (pi []byte)
+}
+
 // ECVRFParams holds shared values across ECVRF implementations.
 // ECVRFParams also has generic algorithms that rely on ECVRFAux for specific sub algorithms.
 type ECVRFParams struct {
 	suite    byte           // Single nonzero octet specifying the ECVRF ciphersuite.
 	ec       elliptic.Curve // Elliptic curve defined over F.
-	fieldLen int            // Length, in bytes, of a field element in F. Defined as 2n in spec.
-	ptLen    int            // Length, in bytes, of an EC point encoded as an octet string.
-	qLen     int            // Length, in bytes, of the prime order of the EC group (Typically ~fieldLen).
+	fieldLen uint           // Length, in bytes, of a field element in F. Defined as 2n in spec.
```

Comment:
It would probably be good to add explicitly that this must be a multiple of 2, as otherwise there might be some odd behavior in other parts of the implementation.